lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/StateMachine/DriverTaskState.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace Org.Apache.REEF.IMRU.OnREEF.Driver.StateMachine
+{
+    /// <summary>
+    /// Driver Task State represents task states from creating a new task, to submitted task, to task running, until task is completed. 
+    /// It also defines the state transition condition from one to another
+    /// All the task states are defined in <see cref="TaskState"></see>
+    /// For the task state transition diagram <see href="https://issues.apache.org/jira/browse/REEF-1327"></see>
+    /// This class wraps current state and provides methods to move from one state to the next state
+    /// </summary>
+    internal sealed class DriverTaskState
```

Comment:
I like option 2. This will also make our tests for state machine simpler, as we can just test individual transitions and don't have to limit ourselves to transitions from current state.